### PR TITLE
Refactor MAISI bundle to monai 1.4

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes # .

### Description
refactor to monai1.4, 
add quality check, 
correct typo
add download script

### Status
**Ready/Work in progress/Hold**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: models/maisi_ct_generative/large_files.yml [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
   - path: "models/mask_generation_diffusion_unet.pt"
-    url: "https://drive.google.com/file/d/11SA9RUZ6XmCOJr5v6w6UW1kDzr6hlymw/view?usp=drive_link"
+    url: "https://drive.google.com/file/d/1yOQvlhXFGY1ZYavADM3N34vgg5AEitda/view?usp=drive_link"
   - path: "configs/candidate_masks_flexible_size_and_spacing_3000.json"
     url: "https://drive.google.com/file/d/1yMkH-lrAsn2YUGoTuVKNMpicziUmU-1J/view?usp=sharing"
   - path: "configs/all_anatomy_size_condtions.json"
     url: "https://drive.google.com/file/d/1AJyt1DSoUd2x2AOQOgM7IxeSyo4MXNX0/view?usp=sharing"
   - path: "datasets/all_masks_flexible_size_and_spacing_3000.zip"
-    url: "https://drive.google.com/file/d/16MKsDKkHvDyF2lEir4dzlxwex_GHStUf/view?usp=sharing"
+    url: "https://drive.google.com/file/d/1AJyt1DSoUd2x2AOQOgM7IxeSyo4MXNX0/view?usp=sharing"
```
Comment (KumoLiu) [inline models/maisi_ct_generative/large_files.yml]: Looks this one should not update.

File: models/maisi_ct_generative/scripts/utils.py [hunk 2]
```diff
@@ -396,34 +683,43 @@ def __call__(self, img: NdarrayOrTensor):
...
-                k.replace("conv.weight", "conv.conv.weight")
-                if "conv.weight" in k
-                else k.replace("conv.bias", "conv.conv.bias")
-            )
-            new_state_dict[new_key] = v
-        else:
-            new_state_dict[k] = v
-    checkpoint_autoencoder = new_state_dict
-    return checkpoint_autoencoder
+def KL_loss(z_mu, z_sigma):
```
Comment (yiheng-wang-nv) [inline models/maisi_ct_generative/scripts/utils.py]: Hi @Can-Zhao , for functions are not used in the bundle, should we keep it?

File: models/maisi_ct_generative/scripts/utils.py [hunk 3]
```diff
@@ -102,68 +163,254 @@ def supress_non_largest_components(img, target_label, default_val=0):
...
+
+    Returns:
+        The instantiated object as defined by the instance_def_key in the parsed configuration.
+    """
+    parser = ConfigParser(vars(args))
+    parser.parse(True)
+    return parser.get_parsed_content(instance_def_key, instantiate=True)
+
+
+def add_data_dir2path(list_files: list, data_dir: str, fold: int = None) -> tuple[list, list]:
```
Comment (yiheng-wang-nv) [inline models/maisi_ct_generative/scripts/utils.py]: there are two `add_data_dir2path` functions in this file

File: models/maisi_ct_generative/large_files.yml (hunk 1 above)
Comment (yiheng-wang-nv) [reply models/maisi_ct_generative/large_files.yml]: HI @KumoLiu , why no need to change? It seems the network is changed and the old version cannot work.

Comment (KumoLiu) [reply models/maisi_ct_generative/large_files.yml]: It's dataset not model weight.

Comment (yiheng-wang-nv) [reply models/maisi_ct_generative/large_files.yml]: I see. This line is duplicate from the above,

Comment (yiheng-wang-nv) [reply models/maisi_ct_generative/large_files.yml]: Hi @Can-Zhao , please confirm it

File: models/maisi_ct_generative/scripts/sample.py [hunk 4]
```diff
@@ -327,13 +496,20 @@ def __init__(
         controllable_anatomy_size,
         image_output_ext=".nii.gz",
         label_output_ext=".nii.gz",
-        quality_check_args=None,
+        real_img_median_statistics="./configs/image_median_statistics.json",
```
Comment (yiheng-wang-nv) [inline models/maisi_ct_generative/scripts/sample.py:499]: Hi @Can-Zhao , here I  think we should use an arg,  and cannot hard code to this path. CI failed and error shows that cannot find this file.
We may need sth like:
"real_img_median_statistics": "$@bundle_root + '/configs/image_median_statistics.json'"

File: models/maisi_ct_generative/large_files.yml (hunk 1 above)
Comment (Can-Zhao) [reply models/maisi_ct_generative/large_files.yml]: I copied the link in tutorial...It seems we have saved this file in more than one google links

Comment (yiheng-wang-nv) [reply models/maisi_ct_generative/large_files.yml]: I switched to the original link (confirmed it still works)

File: models/maisi_ct_generative/scripts/find_masks.py [hunk 5]
```diff
@@ -12,109 +12,144 @@
...
+            if "label_filename" in _item:
+                candidate["label"] = os.path.join(mask_foldername, _item["label_filename"])
 
-        candidates.append(candidate)
+            candidate_masks.append(candidate)
 
-    if len(candidates) == 0 and not check_spacing_and_output_size:
-        raise ValueError("Cannot find body region with given organ list.")
+    if len(candidate_masks) == 0 and not check_spacing_and_output_size:
+        raise ValueError("Cannot find body region with given anatomy list.")
```
Comment (yiheng-wang-nv) [inline models/maisi_ct_generative/scripts/find_masks.py:153]: need to modify the unit test case

File: models/maisi_ct_generative/scripts/sample.py (hunk 4 above)
Comment (yiheng-wang-nv) [reply models/maisi_ct_generative/scripts/sample.py:499]: I pushed a commit to resolve it

File: models/maisi_ct_generative/scripts/find_masks.py (hunk 5 above)
Comment (yiheng-wang-nv) [reply models/maisi_ct_generative/scripts/find_masks.py:153]: I pushed a commit and switched to use anatomy list in `find_closest_masks`

File: models/maisi_ct_generative/scripts/utils.py (hunk 2 above)
Comment (Can-Zhao) [reply models/maisi_ct_generative/scripts/utils.py]: You are right. We should remove it. Thanks!

File: models/maisi_ct_generative/scripts/sample.py (hunk 4 above)
Comment (Can-Zhao) [reply models/maisi_ct_generative/scripts/sample.py:499]: Thank you so much!

File: models/maisi_ct_generative/scripts/find_masks.py (hunk 5 above)
Comment (Can-Zhao) [reply models/maisi_ct_generative/scripts/find_masks.py:153]: Thank you so much!

**Review by yiheng-wang-nv — APPROVED** [review]

**Pull request merged**